services/brig/test/integration/API/OAuth.hs

```diff
@@ -0,0 +1,780 @@
+        [ test m "register" $ testRegisterOAuthClientAccessDeniedWhenDisabled o b,
+          test m "get client info" $ testGetOAuthClientInfoAccessDeniedWhenDisabled o b,
+          test m "create code" $ testCreateCodeOAuthClientAccessDeniedWhenDisabled o b,
+          test m "create token" $ testCreateAccessTokenAccessDeniedWhenDisabled o b,
+          test m "refresh access token" $ testRefreshAccessTokenAccessDeniedWhenDisabled o b
+        ],
+      testGroup
+        "accessing a resource"
+        [ test m "success (internal)" $ testAccessResourceSuccessInternal b,
+          test m "success (nginz)" $ testAccessResourceSuccessNginz b n,
+          test m "insufficient scope" $ testAccessResourceInsufficientScope b,
+          test m "expired token" $ testAccessResourceExpiredToken o b,
+          test m "nonsense token" $ testAccessResourceNonsenseToken b,
+          test m "no token" $ testAccessResourceNoToken b,
+          test m "invalid signature" $ testAccessResourceInvalidSignature o b
+        ],
+      testGroup "refresh tokens" $
+        [ test m "max active tokens" $ testRefreshTokenMaxActiveTokens o db b,
+          test m "refresh access token - success" $ testRefreshTokenRetrieveAccessToken o b,
+          test m "wrong signature - fail" $ testRefreshTokenWrongSignature o b,
+          test m "no token id - fail" $ testRefreshTokenNoTokenId o b,
+          test m "non-existing id - fail" $ testRefreshTokenNonExistingId o b,
+          test m "wrong client id - fail" $ testRefreshTokenWrongClientId b,
+          test m "wrong client secret - fail" $ testRefreshTokenWrongClientSecret b,
+          test m "wrong grant type - fail" $ testRefreshTokenWrongGrantType b,
+          test m "expired token - fail" $ testRefreshTokenExpiredToken o b,
+          test m "revoked token - fail" $ testRefreshTokenRevokedToken b
+        ],
+      testGroup "oauth applications" $
+        [ test m "list applications with account access" $ testListApplicationsWithAccountAccess b,
+          test m "revoke application account access" $ testRevokeApplicationAccountAccess b
+        ]
+    ]
+
+testRegisterNewOAuthClient :: Brig -> Http ()
+testRegisterNewOAuthClient brig = do
+  let newOAuthClient@(NewOAuthClient expectedAppName expectedUrl) = newOAuthClientRequestBody "E Corp" "https://example.com"
+  cid <- occClientId <$> registerNewOAuthClient brig newOAuthClient
+  uid <- randomId
+  oauthClientInfo <- getOAuthClientInfo brig uid cid
+  liftIO $ do
+    expectedAppName @?= ocName oauthClientInfo
+    expectedUrl @?= ocRedirectUrl oauthClientInfo
+
+testCreateOAuthCodeSuccess :: Brig -> Http ()
+testCreateOAuthCodeSuccess brig = do
+  let newOAuthClient@(NewOAuthClient _ redirectUrl) = newOAuthClientRequestBody "E Corp" "https://example.com"
+  cid <- occClientId <$> registerNewOAuthClient brig newOAuthClient
+  uid <- randomId
+  let scope = OAuthScopes $ Set.fromList [ConversationCreate, ConversationCodeCreate]
+  let state = "foobar"
```

Comment:
it's good practice to not have something too common in here, or there is a (theoretical) risk of getting a "foobar" that's actually coming from somewhere else.

```suggestion
  let state = "10b2df8a-9e22-11ed-93eb-47378d658ebd"
```

(or at least hit the keyboard with the fist once, instead of using such a common word.)

(also below.)

Reply:
did you resolve these comments without changing anything, and without explaining why not?  or did you forget to push?